packages/compiler/src/testing/rule-tester.ts

```diff
@@ -42,10 +53,33 @@ export function createLinterRuleTester(
       const diagnostics = await diagnose(code);
       expectDiagnostics(diagnostics, match);
     }
+
+    function applyCodeFix(fixId: string) {
+      return { toEqual };
+
+      async function toEqual(expectedCode: string) {
+        const diagnostics = await diagnose(code);
+        const codefix = diagnostics[0].codefixes?.find((x) => x.id === fixId);
```

Comment:
This is expected to be used only when there is a single diagnostic?

Reply:
Yeah the expectation is you emit a single linter warning and apply the code fix on it

Reply:
If we endup having cases where there is multiple diagnostic always emitted, I think then we can add a extra option to choose which one